app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,26 @@
+class CartsController < ApplicationController
+  before_action :cart_current, only: %i(create)
+  def index
+    @products = {}
+    cart_current.each do |product_detail_id, quantity|
+      product_detail = ProductDetail.find_by(id: product_detail_id)
```

Comment:
Lấy nguyên 1 object làm key luôn ha cậu

Reply:
Dạ, em đưa thế để về dưới view lấy ra mấy thông tin trong sản phẩm á anh. Có cách nào cleả hơn anh cho em xin cái keyword với ạ.

Reply:
Có thể để nguyên id làm key và quantity là value, lúc query chỉ cần get all key của hash đem đi find thì sẽ được câu where in chứ đừng đưa câu query từng item vào vòng lặp, rồi compare cái key đó lúc each ở view với hash ban đầu để get quantity
Cách này có thể sẽ đỡ tốn performance hơn khi hệ thống phình to đấy, cân nhắc nha

Reply:
dạ em sẽ suy nghĩ ạ

Reply:
Nếu find ko ra thì sao bạn nhỏ, vẫn add line 7 ah